src/DataTypes/Serializations/SerializationBool.cpp

```diff
@@ -164,6 +211,22 @@ void SerializationBool::deserializeFromString(IColumn & column, String & input,
         col->insert(false);
     }
     else
-        throw Exception("Invalid boolean value, should be " + settings.bool_true_representation + " or " + settings.bool_false_representation + " controlled by setting bool_true_representation and bool_false_representation.", ErrorCodes::ILLEGAL_COLUMN);
+    {
+        String input_lower = boost::algorithm::to_lower_copy(input);
```

Comment:
I also think we can make hand-crafted code with branches over characters and `toLowerIfAlphaASCII` function to avoid temporary string and copying. We will also remove the unordered map. It will make code less generic but ok.